client/cli/src/lib.rs

```diff
@@ -346,6 +347,42 @@ pub fn init_logger(
 	Ok(())
 }
 
+/// A writer that may writes to `stderr` with colors.
+///
+/// This is used by the logging to kill colors when they are disabled.
+///
+/// If the inner is `false`, the colors will be removed before writing to stderr.
+#[derive(Clone)]
+struct MaybeColorWriter(bool);
+
+impl std::io::Write for MaybeColorWriter {
+	fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+		lazy_static::lazy_static! {
+			static ref RE: Regex = Regex::new("\x1b\\[[^m]+m").expect("Error initializing color regex");
+		}
```

Comment:
You probably could store the regex in `self` and drop lazy_static dependency. Though I don't mind lazy_static as we probably need it anyway.